### ISSUE TITLE
Gas Estimation Error

### ISSUE THREAD

**Issue by TheGreatAxios** [skalenetwork/skaled#1633]
**Describe the bug**
When calling Solidity contract, usage of state after it has been deleted or set to 0/empty results in errors during gas estimation on SKALE Chain

**To Reproduce**
1. Create Solidity smart contract that contains a mapping of address to uint256
2. Create function that sets value to 100 for your address when called
3. Create function that sets value to 0 for your address when called and emits an event that uses the previous balance e.g balances[msg.sender]
5. See error - in ethers it is: **ProviderError: Transaction gas amount is less than the intrinsic gas amount for this transaction type.** when attempting to send transaction (v5.7.1)

**Expected behavior**
The transaction should succeed

**Additional context**
- The transaction can be sent successfully by setting gas limit high enough

```[tasklist]
### Tasks
- [ ] https://github.com/skalenetwork/skaled/issues/1637
```

**Comment by PolinaKiporenko** [comment]
@olehnikolaiev please investigate the root cause

**Comment by PolinaKiporenko** [comment]
after discussion with @DmytroNazarenko @kladkogex @dimalit @olehnikolaiev decided to do quick fix for 2.2 release - https://github.com/skalenetwork/skaled/issues/1637 to unblock customer and work on this issue in 2.3 release

**Comment by oleksandrSydorenkoJ** [comment]
Verified in skaled 3.18.0
partially fixed in https://github.com/skalenetwork/skaled/issues/1637
EstimateGas fix in  https://github.com/skalenetwork/skaled/issues/1600

skale chain breakable-anguished-tania-borealis
```
curl -X POST --data '{"jsonrpc":"2.0","method":"eth_estimateGas","params":[{"from":"0xb04a7f2Cd74Eb624Df4BD8bed8f7034B5Aed89C0", "to":"0x8057Ed7F06585F491Fd3BbC18B796759d74F6B7F", "data":"0xc0fe1af80000000000000000000000000000000000000000000000000000000000000000"}], "id": 1 }' http://1.1.1.1:10387
{"id":1,"jsonrpc":"2.0","result":"0x52c8"}
```
